### PR TITLE
Add detection of {micro,open}shift to handle CNI binary

### PR DESCRIPTION
**Pull request by bn222**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -212,7 +212,7 @@ images-buildx: ## Build all container images necessary to run the whole operator
 	buildah manifest push --all $(DPU_DAEMON_IMAGE)-manifest docker://$(DPU_DAEMON_IMAGE)
 
 .PHONY: images-push
-images-push: test ## Push all container images necessary to run the whole operator
+images-push: ## Push all container images necessary to run the whole operator
```
Comment (wizhaoredhat) [inline Makefile:215]: Some commit conflict here. You did this change in another commit.

File: config/dev/local-images.yaml [hunk 2]
```diff
@@ -19,8 +19,8 @@ spec:
         name: manager
         env:
         - name: DPU_DAEMON_IMAGE
-          value: localhost:5000/dpu-daemon:dev
+          value: wsfd-advnetlab218.anl.eng.bos2.dc.redhat.com:5000/dpu-daemon:dev
```
Comment (wizhaoredhat) [inline config/dev/local-images.yaml:22]: Left over debug.

File: Makefile (hunk 1 above)
Comment (bn222) [reply Makefile:215]: I'll rebase.

**Pull request merged**